lib/opentelemetry.js

```diff
@@ -0,0 +1,186 @@
+// Initialize OpenTelemetry tracer singleton and export
+// the most useful API members, the tracer and some helper functions.
+//
+// The library shouldn't be used directly (although it's not fatal),
+// it is registered as self.apos.telemetry pseudo module (see index.js).
+const api = require('@opentelemetry/api');
+const version = require('../package.json').version;
+
+/** @type {api.Tracer} */
+let tracer;
+
+const AposAttributes = {
+  SCENE: 'apos.scene',
+  TEMPLATE: 'apos.template',
+  EVENT_MODULE: 'apos.event.module',
+  EVENT_NAME: 'apos.event.name',
+  // The module and method targeted by the current operation (see doc module)
+  TARGET_NAMESPACE: 'apos.target.namespace',
+  TARGET_FUNCTION: 'apos.target.function'
+};
+
+module.exports = function (options = {}) {
+  if (!tracer) {
+    tracer = api.trace.getTracer('apostrophe', version);
+
+    // This shouldn't be needed, but having it doesn't hurt
+    if (options.openTelemetryProvider) {
+      api.trace.setGlobalTracerProvider(options.openTelemetryProvider);
+    }
+  }
+
+  /**
+   * Start and return a new span. Optionally provide a parent span or allow the parent
+   * span to be auto-detected.
+   * Use this when the current code block does the tracing and it doesn't expect
```

Comment:
I don't think I understand why we need both this and aposStartActiveSpan.

Reply:
They both are needed. This one only detects the active context and does tracing, without carrying about the next (deeper) active state - it doesn't explicitly wrap other spans. This would be used with e.g. attribute drilling to be passed down for implicitly set the relationships. `startActiveSpan` does both, tracing and activates the nested tracing operation context (explicit usage, it shouldn't be passed down, next span will retrieve the active context). This is related with how span relationships are built with OTel.